demisto_sdk/commands/common/hook_validations/readme.py

```diff
@@ -55,9 +56,15 @@ def is_valid_file(self) -> bool:
 
     def mdx_verify(self) -> bool:
         mdx_parse = Path(__file__).parent.parent / 'mdx-parse.js'
-        # run the java script mdx parse validator
-        _, stderr, is_not_valid = run_command_os(f'node {mdx_parse} -f {self.file_path}', cwd=self.content_path,
-                                                 env=os.environ)
+        with open(self.file_path, 'r') as f:
+            readme_content = f.read()
+        readme_content = self.fix_mdx(readme_content)
+        with tempfile.NamedTemporaryFile('w+t') as fp:
+            fp.write(readme_content)
```

Comment:
why do you read the original file and then write to temporary file?

Reply:
As we modify the original content with the function `fix_mdx` and then we need to write it back as the script works with a file path.

Reply:
also, we got lots of problems with tests writing to the test file itself. If we're using files - we should only read from them,.